libraries/standard/mqtt/src/iot_mqtt_network.c

```diff
@@ -381,7 +381,19 @@ static IotMqttError_t _deserializePublish( _mqttConnection_t * pMqttConnection,
         pOperation->u.publish.pReceivedData = pIncomingPacket->pRemainingData;
         pIncomingPacket->pRemainingData = NULL;
 
-        /* Add the PUBLISH to the list of operations pending processing. */
+        /* Add the PUBLISH to the list of operations pending processing.
+         * Coverity checker USE_AFTER_FREE found a warning when trying to
+         * dereference the 'pMqttConnection' in 'IotMutex_Lock'. For this
+         * warning, freeing the pointer 'pMqttConnection' happens in function
+         * '_sendPuback' invoked at an earlier place in this function.
+         * '_sendPuback' further calls '_IotMqtt_CreateOperation' with 'flags'
+         * with value 0. It takes an invalid path by evaluating 'waitable' as
+         * 'true' by the below expression.
+         * waitable = ( ( flags & IOT_MQTT_FLAG_WAITABLE ) == IOT_MQTT_FLAG_WAITABLE ).
+         * IOT_MQTT_FLAG_WAITABLE is defined as 1.
+         * Hence, this warning is a false positive result and with Coverity
+         * annotation, marking to ignore for future runs. */
+         /* coverity[deref_after_free] */
```

Comment:
This line and 573 look like they're off by a space.

Reply:
those are for 2 different events. deref_after_free and pass_freed_arg

Reply:
I meant these lines have an extra space character at the beginning compared to the previous comment blocks. Is that intentional?

Reply:
ok. got what you meant. It was not off by space though. It was aligned differently. Will fix it